cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,26 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+function ensure_k8s_object {
+  object_type=$1
+  name=$2
+  namespace=$3
+
+  intervals=30
+  timeout=3
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  count=0
+  until _kubectl get $object_type $name $namespace; do
```

Comment:
I think this format is better for such cases:
`for i in {1..$timeout}; do <operations> && break || sleep $interval; done`
You may need to save a return value so you can return it, it depends what is the return value of the `for` when it finishes without `break`.

You should not use `exit` in the function, it exists the whole script and that is to be determined by the caller not by this specific function.

The naming also needs to fit some reasonable well knows naming.
`timeout` is the time in total you are willing to wait for the operation to succeed.
`interval` is the poll accuracy where you re-check the status to exit earlier.